### ISSUE TITLE
Allow SSL connection when using arbitrary server

### ISSUE THREAD

**Issue by ibennetch** [phpmyadmin/phpmyadmin#12355]
2.3) Add setting for using SSL connections in login when $cfg['AllowArbitraryServer']=true
 
When $cfg['AllowArbitraryServer'] is true, the user can enter the server to connect to, but doesn't have option to enable ssl connection.

[I suggest to try ssl connection first and use it by default for non local machine 127.0.0.1/localhost, if it failed try non-ssl connection and show message about it [as in point 2.2 #12354]]
 
Motivations:
* Increase usage of MYSQL TLS connections (avoid sniffing/MITM attacks)

**Comment by ibennetch** [comment]
Suggested by Emanuel Bronshtein

**Comment by LeDistordu** [comment]
+1

**Comment by bline108** [comment]
any progress or plans to implement this? seems pretty important feature for anyone using arbitrary server feature.

**Comment by williamdes** [comment]
> any progress or plans to implement this? seems pretty important feature for anyone using arbitrary server feature.

Thanks for the ping, I assigned this to the next minor version. I think we could add a checkbox to enable SSL for arbitrary connections

**Comment by ThreeDfish** [comment]
I see that this didn't make it into the 5.3.0 milestone.  Has anybody done any work on this?  It seems to me that this is a fairly important security feature.

If I created a pull for this to add a checkbox for SSL for arbitrary connections that would default to checked, what are the chances that it would get accepted?

**Comment by williamdes** [comment]
> I see that this didn't make it into the 5.3.0 milestone. Has anybody done any work on this? It seems to me that this is a fairly important security feature.
> 
> If I created a pull for this to add a checkbox for SSL for arbitrary connections that would default to checked, what are the chances that it would get accepted?

Yes most probably, can you make a pull-request for the master branch?
So this goes into 6.0?
